### PR TITLE
Use edx-drf-extentsions 0.5.0 for JWT authentication support

### PR DESCRIPTION
**Pull request by bderusha**
FYI @clintonb @peter-fogg

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+cryptography==1.3.1
```
Comment (clintonb) [inline requirements/base.txt:1]: This is causing issues. Why was this added @bderusha?

Comment (clintonb) [inline requirements/base.txt:1]: There is no Python 3.5-compatible.

Comment (bderusha) [inline requirements/base.txt:1]: jwt requires it for using asymmetric key algorithms

Comment (bderusha) [inline requirements/base.txt:1]: There is a note on PyPi about using another library.  I'll see if I can get that working:

`If your system doesn’t allow installing cryptography like on Google App Engine, you can install PyCrypto for RSA signatures and ecdsa for ECDSA signatures.`

Comment (clintonb) [inline requirements/base.txt:1]: @bderusha it _may_ be worthwhile to figure out why this build is broken: https://travis-ci.org/edx/configuration/jobs/124581567#L2023-L2034. This is why prompted by investigation. Yesterday, `cryptography` wasn't building on my machine. Now it is! The installation continuously fails on the Travis build, however.
